### PR TITLE
remote: Include file size in hash-mismatch message

### PR DESCRIPTION
**Pull request by Yannic**

### CODE REVIEW

**Comment by jin** [comment]
cc @ulfjack

File: src/main/java/com/google/devtools/build/lib/remote/util/DigestOutputStream.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+import com.google.common.hash.HashFunction;
+import com.google.common.hash.Hasher;
+import java.io.FilterOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+
+/**
+ * An {@link OutputStream} that maintains a {@link Digest} of the data written to it.
+ *
+ * <p>Based on Guava's {@link com.google.common.hash.HashingOutputStream}.
```
Comment (ulfjack) [inline src/main/java/com/google/devtools/build/lib/remote/util/DigestOutputStream.java:30]: s/Based on/Similar to/ and also add that it returns a Digest object that's specific to the remote execution protocol.

File: src/main/java/com/google/devtools/build/lib/remote/util/Utils.java [hunk 2]
```diff
@@ -173,13 +173,13 @@ public static String grpcAwareErrorMessage(IOException e) {
...
+  public static void verifyBlobContents(Digest expected, Digest actual) throws IOException {
+    if (!expected.equals(actual)) {
       String msg =
           String.format(
-              "An output download failed, because the expected hash"
-                  + "'%s' did not match the received hash '%s'.",
-              expectedHash, actualHash);
+              "Output download failed: Expected to download a blob of %d bytes with hash '%s', "
+                  + "got %d bytes with '%s'.",
+              expected.getSizeBytes(), expected.getHash(), actual.getSizeBytes(), actual.getHash());
```
Comment (ulfjack) [inline src/main/java/com/google/devtools/build/lib/remote/util/Utils.java:182]: Personally, I'd prefer to format it like digests elsewhere as `<hash>/<size>`. WDYT?

**Comment by ulfjack** [comment]
LGTM! Looking forward to seeing this merged. \o/

File: src/main/java/com/google/devtools/build/lib/remote/util/Utils.java (hunk 2 above)
Comment (Yannic) [reply src/main/java/com/google/devtools/build/lib/remote/util/Utils.java]: SGTM, done

File: src/main/java/com/google/devtools/build/lib/remote/util/DigestOutputStream.java (hunk 1 above)
Comment (Yannic) [reply src/main/java/com/google/devtools/build/lib/remote/util/DigestOutputStream.java]: done

**Review by coeuvre — APPROVED** [review]
LGTM. Thanks!

**Pull request closed without merging**